lightningd/lightningd.c

```diff
@@ -251,26 +409,40 @@ static void init_txfilter(struct wallet *w, struct txfilter *filter)
 	}
 }
 
+/*~ The normal advice for daemons is to move into the root directory, so you
+ * don't prevent unmounting whatever filesystem you happen to start in.
+ *
+ * But we define every path relative to our (~/.lightning) data dir, so we
+ * make sure we stay there.
+ */
 static void daemonize_but_keep_dir(struct lightningd *ld)
 {
 	/* daemonize moves us into /, but we want to be here */
 	const char *cwd = path_cwd(NULL);
 
+	/*~ SQLite3 does NOT like being open across fork(), a.k.a. daemonize() */
 	db_close_for_fork(ld->wallet->db);
 	if (!cwd)
 		fatal("Could not get current directory: %s", strerror(errno));
 	if (!daemonize())
 		fatal("Could not become a daemon: %s", strerror(errno));
 
-	/* Move back: important, since lightning dir may be relative! */
+	/*~ Move back: important, since lightning dir may be relative! */
 	if (chdir(cwd) != 0)
 		fatal("Could not return to directory %s: %s",
 		      cwd, strerror(errno));
 
 	db_reopen_after_fork(ld->wallet->db);
+
+	/*~ Why not allocate cwd off tmpctx?  Probably because this code predates
+	 * tmpctx.  So we free manually here. */
```

Comment:
I thought the it's more because the global `tmpctx` is an event loop concept and this is outside of the loop.

Reply:
Yeah, but it's before the loop; we tend to allow that elsewhere...